### DISCUSSION TITLE
Try to understand the difference between event.name and logrecord.event.name and where the event.id is represented in otel

### DISCUSSION THREAD

**Discussion by HHobeck** [open-telemetry/opentelemetry-dotnet#5804] [Q&A] [answered]
Hi there.

In the [readme.md](https://github.com/open-telemetry/opentelemetry-dotnet/blob/1b3f1894d95fd01fe475245ec60ef273d5f77bb6/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md?plain=1#L658) the following feature switch is documented:

>   * `OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES`
> 
>     When set to `true`, it enables export of `LogRecord.EventId.Id` as
>     `logrecord.event.id` and `LogRecord.EventId.Name` as `logrecord.event.name`.
> 
>     Added in `1.7.0-alpha.1`.

When I create an event log in dotnet core 8.0 with OpenTelemetry.Exporter.OpenTelemetryProtocol (1.9.0) I get the following result:
```csharp
Environment.SetEnvironmentVariable(
    "OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES", "true"
);
logger.LogInformation(new EventId(4711, "JustATest"), "JustATest log");
```

```json
{
  "resourceLogs": [
    {
      "resource": {
        "attributes": [
          ..
        ]
      },
      "scopeLogs": [
        {
          "scope": { "name": "MyApplication" },
          "logRecords": [
            {
              "timeUnixNano": "1724672694064471500",
              "observedTimeUnixNano": "1724672694064471500",
              "severityNumber": 9,
              "severityText": "Information",
              "body": { "stringValue": "JustATest log" },
              "attributes": [
                {
                  "key": "logrecord.event.id",
                  "value": { "intValue": "4711" }
                },
                {
                  "key": "logrecord.event.name",
                  "value": { "stringValue": "JustATest" }
                }
              ],
              "flags": 1,
              "traceId": "1fc1034d6359bef1373fa7d6e741e1d5",
              "spanId": "2834927eca22c371"
            }
          ]
        }
      ]
    }
  ]
}
```

My questions:
1. Is the resulting json output of `logrecord.event.id` and `logrecord.event.name` correct? I would expect just `event.id` and `event.name` because we are already in the logRecords scheme!
1. I cannot find the event definition in the otel convention. The only reference I have found is [Event definition](https://opentelemetry.io/docs/specs/semconv/general/events/). But here only the `event.name` is documented not `event.id`.
1. The status is exprimental. How long does it take to get a final specification?

**Comment by cijothomas** [comment, marked as answer]
OTel's Log data model does not have a place to put ILogger's EventId and Name, so they are now exported as attributes. Since there is no standardized attribute name spec-ed out today, the current ones("logrecord.*") were chosen, and marked experimental. The final name could be something else, or it could be a top-level field in OTel LogRecord or something else. 

No firm ETA on this, you can track progress here : https://github.com/open-telemetry/opentelemetry-dotnet/issues/5462

**Reply by cijothomas** [reply]
https://github.com/open-telemetry/opentelemetry-dotnet/issues/4404 Also see this.

**Reply by HHobeck** [reply]
Thank you providing this information. I hope this topic will go to the right path in the near future.
